tests/device_tests/solana/construct/custom_constructs.py

```diff
@@ -0,0 +1,103 @@
+from base58 import b58decode, b58encode
```

Comment:
When possible, please also provide type-hints to these helper functions (no need to describe some complex objects, but things like `bytes`, `str` or `int` are quick and easy to add)

Reply:
https://github.com/trezor/trezor-firmware/pull/3359/commits/27c9271891a38edf27374989bd952b5065034b19.

Reply:
on my system `from base58` is not recognized.

however we have base58 encoding in `trezorlib.tools`

Reply:
I'm not sure exactly what the problem is but decoding `11111111111111111111111111111111` with base58 from trezorlib yields 33 bytes while decoding with the `base58` package yields 32 bytes.

I expect it has to do with [this line](https://github.com/trezor/trezor-firmware/blob/02e6766e2daabb0099c954ecba4a0cd89673aa20/python/src/trezorlib/tools.py#L181) and `nPad` being 1 but I don't know the specifics of base58 so I don't know why it's there or what might be wrong.

Actually with Trezorlib base58 `b58encode(b58decode("11111111111111111111111111111111"))` does not equal `"11111111111111111111111111111111"`. (there's the extra byte)

Reply:
well that's fun! this should help: 0bec4113b051bcab056d1208d1622e5d202a18a9

Reply:
Thanks! Cherry picked and used in https://github.com/trezor/trezor-firmware/pull/3359/commits/65397e0bc4830070a1b57c06d8e727dc97778d82.